viewing_party/main.py

```diff
@@ -0,0 +1,251 @@
+    unique_friend_list = []
+    for x in friends_bank:
+        if x not in unique_friend_list:
+            unique_friend_list.append(x)
+#comparing friends seen movies with user's movies
+    teo =[]
+    for x in unique_friend_list:
+        if x not in user_bank:
+            teo.append(x)
+#returning a list of dictionaries \
+# representing a list of movies
+    unseen_movies = {}
+    
+    for x in range(len(teo)):
+        unseen_movies["title"] = teo[x]
+        new_movie_for_user.append(unseen_movies)
+        unseen_movies = {}
+    return new_movie_for_user
+
+#WAVE FOUR
+
+def get_available_recs(user_data):
+    new_rec = get_friends_unique_watched(user_data)
+    friends_bank_host =[]
+    friends_bank_title =[]
+    fin_mov =[]
+    user_host =[]
+#Retrieve Value in dictionary in new_rec
+    for x in new_rec:
+        for a,b in x.items():
+            fin_mov.append(b)
+#Finding host in recommended movies 
+    for y in user_data["friends"]:
+        for z in y["watched"]:
+            for key,value in z.items():
+                if z["title"] in fin_mov:
+                    friends_bank_host.append(z["host"])
+#Removing duplicate movies
+    friends_bank_host_unique =[]
+    for x in friends_bank_host:
+        if x not in friends_bank_host_unique:
+            friends_bank_host_unique.append(x)
+#Creating a list of dictionaries   
+    movie_rec ={}   
+    movie_rec_list =[]  
+    for x in range(len(fin_mov)):
+        movie_rec["title"] = fin_mov[x]
+        movie_rec["host"] = friends_bank_host_unique[x]
+        movie_rec_list.append(movie_rec)
+        movie_rec ={}
+#Comparing host of rec movies is in the user's subscription    
+    final_list_host = []
+    for x in movie_rec_list:
+        for key, value in x.items():
+            if value in user_data["subscriptions"]:
+                final_list_host.append(value)
+#Appends movie from list of dictionaries
+    final_list =[]
+    for x in movie_rec_list:
+        if x["host"] in final_list_host:
+            final_list.append(x)
+    
+    return final_list
+
+#WAVE FIVE
+#1\
+# Using previous functions, finding pop.genera of user
+def get_new_rec_by_genre(user_data):
+    most_genre = get_most_watched_genre(user_data)
+    movie_rec = get_friends_unique_watched(user_data)
+    genre_rec_unq =[]
+    genre_rec = []
+#if movie list or favorite genre is empty return emptylist
+    if len(movie_rec) == 0:
+        return genre_rec_unq
+    elif most_genre == None:
+        return genre_rec_unq
+    else:
+        for y in user_data["friends"]:
+            for z in y["watched"]:
+                for key,value in z.items():
+                    if z["genre"] in most_genre:
+                        genre_rec.append(z)
+#Removing duplicate genres
+        genre_rec_unq = []
+        for x in genre_rec:
+            if x not in genre_rec_unq:
+                genre_rec_unq.append(x)
+        return genre_rec_unq
+
+
+def get_rec_from_favorites(user_data):
```

Comment:
Question could you use a prior method to get the friends watched movies, or could you make a helper method to break that part up?